scripts/scr_add_man/scr_add_man.gml

```diff
@@ -1,47 +1,184 @@
+            obj_ini.w[company,marine_number] = new_weapon;
+	   };
+		static armour = function(){ 
+			return obj_ini.armour[company,marine_number];
+		};	   
+       static update_armour = function(new_armour){
+            obj_ini.armour[company,marine_number] = new_armour;
+			get_unit_size(); //every time armour is changed see if the marines size has changed
+	   };	
+		static corruption = function(){ 
+			return obj_ini.chaos[company,marine_number];
+		};	   
+       static update_corruption = function(new_corruption){
+            obj_ini.chaos[company,marine_number] = new_corruption;
+	   };	
+		static specials = function(){ 
+			return obj_ini.spe[company,marine_number];
+		};	   
+       static update_specials = function(new_specials){
+            obj_ini.spe[company,marine_number] = new_specials;
+	   };
+	   	static mobility_item = function(){ 
+			return obj_ini.mobi[company,marine_number];
+		};	   
+       static update_mobility_item = function(new_mobility_item){
+            obj_ini.mobi[company,marine_number] = new_mobility_item;
+			get_unit_size(); //every time mobility_item is changed see if the marines size has changed
+	   };	
+	   	static race = function(){ 
+			return obj_ini.race[company,marine_number];
+		};	
+		static role = function(){
+			return obj_ini.role[company,marine_number];
+		};
+		static update_role = function(new_role){
+			obj_ini.role[company,marine_number]= new_role;
+			array_append(role_history ,[obj_ini.wep1[company,marine_number], obj_controller.turn])
+		}		
+		
+		static get_unit_size = function(){
+			var r = role();
+			var arm = armour();
+		    var sz = 1;
+		    if (r!=""){
+				var bulky_armour = ["Terminator Armour", "Tartaros"]
+		        if (string_count("Dread",arm)>0) {sz+=5;}else if (array_contains(bulky_armour,arm)){sz +=1};
+		        if (r="Rhino") {sz=10;}
+		        else if (r="Predator") {sz=10;}
+		        else if (r="Land Raider") {sz=20;} 
+		        else if (r="Land Speeder") {sz=5;}
+		        else if (r="Whirlwind") {sz=10;}
+		        else if (r="Harlequin Troupe") {sz=5;}
+				else if (r="Chapter Master"){sz+=1;}
+				
+				var mobi =  mobility_item();
+				if (mobi == "Jump Pack"){
+					sz += 1;
+				}
+		        size =sz;
+		    }
+	
+			size = 0;			
+		}
+		
+		static marine_location = function(){
+			var location_id,location_name;
+			var location_type = obj_ini.wid[company,marine_number];
+			if ( location_type > 0){ //if marine is on planet
+				location_id = location_type; //planet_number marine is on
+				location_type = "planet"; //state marine is on planet
+				location_name = obj_ini.loc[company,marine_number]; //system marine is in
+			} else {
+				location_type = "ship"; //marine is on ship
+				location_id = obj_ini.lid[_company, _unit]; //ship array location
+				location_name = obj_ini.ship_location[location_id]; //location of ship
+			}
+			return [location_type,location_id ,location_name];
+		};
+		
+		static load_marine =function(ship){
```

Comment:
@GalacticChimp the standardised struct method for loading a marine to a ship either from a planet or from another ship (will handle wither) will be callable via <<marine>>.load_marine(<<ship>>).

Ignore the fact that all this is being contained in the scr_add_man script, i just prefer to work in  existing files and then split the scripts up later....Idk why.